[ashleyclx] iP

ashleyclx opened the pull request:
# Yapper
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Yapper frees your mind of having to remember things you need to do. It's,

* text-based tiktok zoomer brainrot speech
* easy to learn
* ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [X] Managing tasks
- [X] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

##
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ashleyclx committed:
Level-0

Rename, Greet, Exit

ashleyclx committed:
Changed text personality and logo

ashleyclx committed:
Changed bye text and changed to function

ashleyclx committed:
Created hello function

ashleyclx committed:
Level-1: Echo

Added scanner, echo text and changed some text

ashleyclx committed:
Level-2: Add, List

ashleyclx committed:
Level-3: Mark as Done and Task Class

Created a Task class and made mark as done, undone functions. Override the toString() methods Task

ashleyclx committed:
Added Deadline, Event, Todo Class

Added the 3 subclasses. changed Task to abstract class, changed the getStatusSymbol in Task, toString() methods and print statements

ashleyclx committed:
Changed Duke to accept Deadline, Event and Todo

Edited program logic, print statements.
addTask method now creates Todo, Deadline and Event Tasks.

ashleyclx committed:
Changed printed text

To fit bot personality

ashleyclx committed:
Added test into input.txt and EXPECTED.txt

ashleyclx committed:
Bug fixes

ashleyclx committed:
Added Error handling messages

catches invalid commands, arguments, index, formatting

ashleyclx committed:
Changed tasks from Array to Array list

Updated all methods

ashleyclx committed:
Added delete

added delete command, edited error messages

ashleyclx committed:
Added enum Command and Task.ID

ashleyclx committed:
Update Duke Class to throw DukeExceptions

Handles exceptions as error messages.

Code is bloated by excessive print statements, and should be handled properly by catching exceptions.

Integrate DukeException class in place of print statments. print LINE statements have been moved out of main code logic to reduce bloat also.

ashleyclx committed:
Encapsulate main program flow into parse method

Main program flow of parsing commanf in main.

Bloated main function and hard readability.

Created parseCommand method to handle main assigning of methods and created list, mark, unmark, delete helper methods to encapsulate program.

Pa

ashleyclx committed:
Reformat Duke.java

Reformat to follow java coding standards.

ashleyclx committed:
Reformat long lines in Duke.java

Break up lines longer than 120 chars

ashleyclx committed:
Create FileManager and Parser class

Moved out addTask and parseCommand from Main to Parser. Overloaded Tasks constructers and addTask. Made a loadData method in FileManager and parseData method in Parser.

Changed access modifier of tasks array to protected so that Parser can access it. Possibly to shift into a Storage class in the future.

ashleyclx committed:
Add saveTasks to FileManager

Adds the ability to save tasks after leaving the program.

ashleyclx committed:
Merge branch 'branch-Level-7'

ashleyclx committed:
Remove unnecessary import statements

ashleyclx committed:
Change Task date to type LocalDate

Affects the Task subclasses, Parser and FileManager

ashleyclx committed:
Add 4 Classes for enhanced OOP

FileManager Class to handle loading and storing data onto hard drive.

Parser handles parsing of user input.

TaskList handles operations with the task list.

Ui prints different messages in response to the user input.

ashleyclx committed:
Split files into packages

Files are split into yapper, exception and task packages.

ashleyclx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ashleyclx committed:
Update main class in build.gradle

ashleyclx committed:
Add junit tests for Ui and Tasks

Tests Ui, Todo, Deadline and Event classes

ashleyclx committed:
Update build.gradle for .jar creation

Changes the .jar name and version

ashleyclx committed:
Add checkstyle to gradle

ashleyclx committed:
Fix coding standards violations

ashleyclx committed:
Add Javadocs to classes

All classes have JavaDocs written in except for TaskList as there may be changes to how it operates soon.

ashleyclx committed:
Add find command to Yapper

Find command added to commands, parser, tasklist, ui that utilises String.contains() in Task.

Changed most static methods to instance creating new instances of UI, Parser, TaskList, FileManager in Yapper.run().

ashleyclx committed:
Merge branch 'branch-A-CodingStandard'

ashleyclx committed:
Fix coding standards

ashleyclx committed:
Merge branch 'branch-A-JavaDoc'

ashleyclx committed:
Add JavaDocs to TaskList class

whitesnowx reviewed (commented):
Looks really goodd
Naming conventions are well done, especially ID which could be easily named wrongly

iynixil commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,77 @@
+        return isDone ? "X" : " ";
+    }
+
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void unmark() {
+        isDone = false;
+    }
```
`markAsDone` and `unmark` seem to have inconsistent naming; maybe you could change either to `mark` or `unmarkAsUndone` or anything else you might find more suitable 🤗

iynixil commented on src/main/java/yapper/Parser.java:
```diff
@@ -0,0 +1,317 @@
+     *
+     * @return String representation of task list data to be saved to the local file.
+     */
+    public String parseToData() {
```
slight nit; perhaps the names for `parseData()` and `parseToData()` can be made clearer e.g `parseInputData()`, `parseOutputData()`?

iynixil commented on src/main/java/yapper/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+        this.ui = ui;
+    }
+
+    public int listSize() {
```
slight nit; should this be named `getListSize()` instead?

iynixil commented on src/main/java/yapper/Ui.java:
```diff
@@ -0,0 +1,133 @@
+        mainTasks.setUi(this);
+    }
+
+    public static String indent() {
+        return INDENT;
+    }
+
+    public static String line() {
```
maybe these functions could be named `showIndent()` and `showLine()` instead?

iynixil commented on src/main/java/yapper/Ui.java:
```diff
@@ -0,0 +1,133 @@
+    /**
+     * Greets the user with an introductory message.
+     */
+    public static void hello() {
+        String logo =
+                "       :::   :::           :::        :::::::::       :::::::::       ::::::::::       :::::::::\n"
+                + "      :+:   :+:         :+: :+:      :+:    :+:      :+:    :+:      :+:              :+:    :+:\n"
+                + "      +:+ +:+         +:+   +:+     +:+    +:+      +:+    +:+      +:+              +:+    +:+\n"
+                + "      +#++:         +#++:++#++:    +#++:++#+       +#++:++#+       +#++:++#         +#++:++#:\n"
+                + "      +#+          +#+     +#+    +#+             +#+             +#+              +#+    +#+\n"
+                + "     #+#          #+#     #+#    #+#             #+#             #+#              #+#    #+#\n"
+                + "    ###          ###     ###    ###             ###             ##########       ###    ###\n\n";
+        System.out.print(LINE + "    What's poppin' fam, it's ya boi\n\n" + logo
+                + "    Hit me up with those deets and let's vibe together!\n" + LINE);
+    }
+
+    /**
+     * Sends a message for terminating the program.
+     * Terminates the program by setting a flag hasEnded.
+     */
+    public void bye() {
```
maybe these could be named `showHello()` and `showBye()`? as long as a verb is included 👍

iynixil reviewed (commented):
LGTM! minor nits to fix 🤗

meowwtama commented on src/main/java/yapper/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+        tasks = new ArrayList<>();
+    }
+
+    public void setUi(Ui ui) {
```
where javadocs

meowwtama commented on src/main/java/yapper/Ui.java:
```diff
@@ -0,0 +1,133 @@
+                + "      +#++:         +#++:++#++:    +#++:++#+       +#++:++#+       +#++:++#         +#++:++#:\n"
+                + "      +#+          +#+     +#+    +#+             +#+             +#+              +#+    +#+\n"
+                + "     #+#          #+#     #+#    #+#             #+#             #+#              #+#    #+#\n"
+                + "    ###          ###     ###    ###             ###             ##########       ###    ###\n\n";
```
This logo is simply astounding.  With every glance, it resonates with purpose, leaving a lasting impression of creativity and sophistication.

meowwtama commented on src/main/java/yapper/Parser.java:
```diff
@@ -0,0 +1,317 @@
+            } catch (DateTimeParseException e) {
+                throw new YapperException("Error in the save files 5");
+            }
+
```
extra line space here

meowwtama commented on src/main/java/yapper/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+
+/**
+ * TaskList that handle operations on a list of Task objects.
+ *
```
extra line here

meowwtama reviewed (commented):
Overall, I found your code easy to read! Some small issues i noted!

ashleyclx committed:
Add GUI to Yapper

Yapper employed a text-based UI

Yapper now employs a GUI via JavaFX and FXML

Removed text based implementation, introduced Launcher, Main, MainWindow and DialogBox classes. Text has been edited to no longer use manual indents and Ui class no longer have hasEnded flag resulting in system no longer automatically exiting on bye command

ashleyclx committed:
Add assertions to Parser and TaskList

Parser and TaskList classes have no checks that their references to FileManager and Ui instances are properly assigned.

This can lead to a possible NullPointerException in the future if the way the references are set changes.

Add assertions that references are not null before calling them in methods.

ashleyclx committed:
Improve code quality of Parser and other classes

Parser has some nested logic, and unclear boolean flags and method names.

Results in longer methods having bad readability.

* Add clear names to boolean flags
* Add guard clauses
* Rename method names to make more sense
* Correct some coding standard violations and make naming more consistent in other classes

ashleyclx committed:
Change TaskList operations to utilise Streams

TaskList currently does not utilise streams.

Streams can help improve processing speed.

Add stream operations in listTasks and find methods.

ashleyclx committed:
Merge pull request #2 from ashleyclx/branch-A-Assertions

Add assertions to Parser and TaskList

ashleyclx committed:
Merge branch 'master' into branch-A-CodeQuality

ashleyclx committed:
Merge branch 'master' into branch-A-Streams

ashleyclx committed:
Merge pull request #3 from ashleyclx/branch-A-CodeQuality

Improve code quality of Parser and other classes

ashleyclx committed:
Merge pull request #4 from ashleyclx/branch-A-Streams

Change TaskList operations to utilise Streams

ashleyclx committed:
Add a help command

No command to know what commands are available

Can learn what are the commands and formats

Add a help command that lists out the commands

ashleyclx committed:
Enhance GUI

ashleyclx committed:
Update GUI title

ashleyclx committed:
Update help message personality

ashleyclx committed:
Add Ui.png

ashleyclx committed:
Update README.md

ashleyclx committed:
Extract methods in Parser

Creates parseHasNoArgumentException, parseMarkUnmarkDelete, parseTodo, parseDeadline, parseEvent methods to improve readability

ashleyclx committed:
Extract methods in parseDataToTask in Parser

ashleyclx committed:
Add JavaDocs

ashleyclx committed:
Fix test cases

ashleyclx committed:
Fix title of user guide

Fixed error that left Duke as title instead of Yapper
